src/main/java/com/bmw/hmm_lib/ViterbiAlgorithm.java

```diff
@@ -157,7 +159,9 @@ private boolean hmmBreak(Map<S, Double> message) {
     private ForwardStepResult forwardStep(HmmProbabilities<S, O> hmmProbabilities,
             TimeStep<S, O> prevTimeStep, TimeStep<S, O> curTimeStep, Map<S, Double> message) {
         final ForwardStepResult result = new ForwardStepResult(curTimeStep.candidates.size());
-        assert( !prevTimeStep.candidates.isEmpty());
```

Comment:
I prefer the assertion here. Candidates cannot be empty here, because in this case an HMM break would have occurred and checked before. However, one could give this reason in the assertion message.